src/util/Util.ts

```diff
@@ -142,6 +142,13 @@ const DidMatchFromJwksUri = (
   return jwksUri.includes(issuerDid);
 };
 
+const compareKidWithId = (kid: string, elem: VerificationMethod): boolean => {
+  if (kid.includes("did:") || kid.startsWith("#")) {
+    return elem.id === kid;
+  }
```

Comment:
Heads up: if `kid = #123`and `elem = [{id: did:key:123#123}]` this would fail, right?